Backport of #253

fselmo opened the pull request:
### What was wrong?

Backport of #253 into `v0.11.x` branch. All were successful cherry-pick commits without conflicts.

- [x] Add entry to the [release notes](https://github.com/ethereum/eth-account/blob/main/newsfragments/README.md)

#### Cute Animal Picture

<img width="1277" alt="Screenshot 2024-03-19 at 09 53 18" src="https://github.com/ethereum/eth-account/assets/3532824/e45f97ca-2cad-422c-ad7e-3f631b2a6c69">

fselmo committed:
Re-organize typed transaction file structure + add blob txns:

- Re-organize typed transactions as a directory with each `_TypedTransactionImplementation`
  as its own file.
- Add support for `BlobTransaction` (type = 3).
- Add relevant tests.

Also:

- Normalize transaction dict for ``TypedTransaction.as_dict()`` so that lists or tuples are converted to tuples for consistency and ease of comparison.

(cherry picked from commit c9da2eeda5b8eb48695823bf05f3f69f368908a6)

fselmo committed:
Implement extended blob rlp logic for 4844 definition:

- When a blob transaction is signed, it needs to contain the blob information
  along with commitments and proofs for the consensus layer. We need to be
  able to serialize and deserialize this information.
- Use different serializers for execution layer style `TransactionPayloadBody`
  blob transactions and for `PooledTransaction` blob transactions.

(cherry picked from commit 17fd5ac378935eb793acb8cf4a714313f101093e)

fselmo committed:
Add more testing around blob transactions:

- Add some testing around blob transaction `blobVersionedHashes` validation
  when present in tx dict compared to the computed values from blobs.
- Add testing around the computing of commitment, proofs, and versioned hashes.
- Add testing around the blob count limits for a blob carrying transaction.

(cherry picked from commit a4c8793f5acd37d5b96b216791299de8e080d2b4)

fselmo committed:
UX improvements to transaction type interpretation

(cherry picked from commit 452b17b8987389d69b7341f6904072aea756531f)

fselmo committed:
Add kzg trusted setup to manifest

(cherry picked from commit b515303e3714442cc3044d1ab8278970d5818fdf)

fselmo committed:
newsfragment for #256

kclowes commented on eth_account/_utils/typed_transactions/base.py:
```diff
@@ -0,0 +1,261 @@
+                )
+                for blob, commitment in zip(self.blobs, self.commitments)
+            ]
+        return self._proofs
```
Feel free to take or leave, but maybe this section warrants its own file?

fselmo replied:
💯 I do like that

kclowes commented on tests/core/_test_data/blob_data_1.txt:
Lazy question - did not check the tests... Do we need to add these `txt` files to the Manifest too?

fselmo replied:
I sanity checked the `/dist` and everything in `tests/` looks to be included, including these test data files, so I think we're good here.

kclowes reviewed (approved):
I left some take or leave comments, but looks good!

fselmo committed:
Address comments on PR #256

- Organize ``blob_data_models.py`` a directory below ``blob_transactions`` due to issues with circular dependencies.
